storage-proofs-update/src/vanilla.rs

```diff
@@ -252,6 +254,88 @@ where
     }
 }
 
+// Computes all `2^h` rho values for the given `phi` in the given range. Each rho corresponds to
+// a `high` value, where `high` is the `h` high bits of a node-index.
+#[derive(Debug)]
+pub struct Rhos {
+    rhos: HashMap<usize, Fr>,
```

Comment:
No `Vec`? I believe that this hashmap's keys are the integers `0..2^h`, so I think that this hashmap is just adding a layer of abstraction on top of indexing into a vector, i.e. `rhos_vec[high_bits]` is replaced with `rhos_hashmap[high_bits]`?

Unless `Rhos` will be used slightly differently than `fn rhos` is used now (where all `2^h` rhos are generated at once in a vector and the vector elements are referenced throughout vanilla encoding).

Reply:
Nvm, I see below that it is not guaranteed that the keys are `0..2^h`.